lib/smoke_test.rb

```diff
@@ -0,0 +1,69 @@
+require 'net/http'
+require 'json'
+require 'openssl'
+
+class SmokeTest
+  def initialize
+    secrets         = Rails.application.secrets
+    @urls           = secrets.smoke_test[:urls]
+    @http_header    = secrets.smoke_test[:http_header]
+    @api_token      = secrets.smoke_test[:api_token] || ""
+    @slack_uri      = secrets.smoke_test[:slack_uri]
+    @slack_token    = secrets.smoke_test[:slack_token]
+    @slack_room     = secrets.smoke_test[:slack_room]
+    @slack_username = secrets.smoke_test[:slack_username]
+    @slack_emoji    = secrets.smoke_test[:slack_emoji]
+  end
+
+  def test_endpoints
+    message = "SmokeTest results: "
```

Comment:
Personally, I would make `messages` an array, then you can append each message and at the end, you can join them all together with a `\n` character to render them on a new line